src/test/java/com/yegor256/tojos/MnSynchronizedTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.yegor256.tojos;
+
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Test case for {@link MnSynchronized}.
+ *
+ * @since 0.3.0
+ */
+class MnSynchronizedTest {
+
+    /**
+     * Number of threads.
+     */
+    static final int THREADS = 5;
+
+    /**
+     * The mono under test.
+     */
+    private Mono shared;
+
+    /**
+     * The executor.
+     */
+    private ExecutorService executor;
+
+    /**
+     * An additional rows.
+     */
+    private Collection<Map<String, String>> additional;
+
+    /**
+     * The latch.
+     */
+    private CountDownLatch latch;
+
+    @BeforeEach
+    final void setUp(@TempDir final Path temp) {
+        this.shared = new MnSynchronized(new MnJson(temp.resolve("bar/baz/a.json")));
+        this.executor = Executors.newFixedThreadPool(MnSynchronizedTest.THREADS);
+        this.latch = new CountDownLatch(1);
+        this.additional = rowsByThreads();
+    }
+
+    @Test
+    final void writesConcurrently() throws InterruptedException {
+        for (int trds = 1; trds <= MnSynchronizedTest.THREADS; ++trds) {
+            this.executor.submit(
+                () -> {
+                    this.latch.await();
+                    final Collection<Map<String, String>> increased = this.shared.read();
+                    increased.addAll(this.additional);
+                    this.shared.write(increased);
```

Comment:
@volodya-lombrozo same problem here, the test fails unless I do `synchronized(this.shared)` in this place, but if I do it here(83-85 lines), then MnSynchronized makes no sense at all

Reply:
@volodya-lombrozo but on my local machine the tests don't work, I'm a little confused, btw we can run `mvn qulice:check` without the `-Pqulice` flag, as far as i remember